scipy/stats/morestats.py

```diff
@@ -2692,15 +2692,13 @@ def circmean(samples, high=2*pi, low=0, axis=None):
 
     """
     samples, ang = _circfuncs_common(samples, high, low)
-    res = angle(np.mean(exp(1j * ang), axis=axis))
-    mask = res < 0
+    S = sin(ang).mean(axis=axis)
+    C = cos(ang).mean(axis=axis)
```

Comment:
quite interesting, for the mean estimation `arr.sum()/N` is about 10% faster than `arr.mean()`. However, as the `sin` and `cos` functions are orders of magnitude slower, this gain is lost in variation of times measured by `timeit`. 

```
In [2]: %timeit -n 100 circmean2(p)
100 loops, best of 3: 18.1 ms per loop

In [3]: %timeit -n 100 circmean2(p)
100 loops, best of 3: 18.3 ms per loop

In [4]: %timeit -n 100 circmean(p)                  
^[[A100 loops, best of 3: 18.2 ms per loop

In [5]: %timeit -n 100 circmean(p)
100 loops, best of 3: 18.2 ms per loop
```